src/renderer/sw_engine/tvgSwFill.cpp

```diff
@@ -820,14 +837,16 @@ bool fillGenColorTable(SwFill* fill, const Fill* fdata, const Matrix* transform,
 
     fill->spread = fdata->spread();
 
-    if (ctable) {
-        if (!_updateColorTable(fill, fdata, surface, opacity)) return false;
-    }
-
     if (fdata->type() == Type::LinearGradient) {
-        return _prepareLinear(fill, static_cast<const LinearGradient*>(fdata), transform);
+        if (_prepareLinear(fill, static_cast<const LinearGradient*>(fdata), transform)) {
```

Comment:
```
if (!_prepareLinear(fill, static_cast<const LinearGradient*>(fdata), transform)) return false;
```

Reply:
I understand we don;t want to repeat the same code. so I applied, but I removed this part:
```
    //LOG: What type of gradient?!
    return false;
 ```